test/python/circuit/library/test_grover_operator.py

```diff
@@ -106,12 +106,11 @@ def test_custom_zero_reflection(self):
             self.assertGroverOperatorIsCorrect(grover_op, oracle)
 
         with self.subTest('circuits match'):
-            expected = QuantumCircuit(*grover_op.qregs)
+            expected = QuantumCircuit(*grover_op.qregs, global_phase=np.pi)
```

Comment:
I think this isn't correct on `stable/0.16`, the global phase of pi was added to the grover operator in #5517: https://github.com/Qiskit/qiskit-terra/pull/5517/files#diff-5cc16ff691e55c7c0e9d381f48c31157dcc945f3107f4e6ea246b2a2bcce986e we should check with @Cryoris if that needs to be backported too or not

Reply:
I think we can include that phase change from 5517, since then the Grover operator fits the definition from the papers 🙂

Reply:
ok, I'll add that here then